build-system/tasks/runtime-test.js

```diff
@@ -541,96 +546,123 @@ function runTests() {
   const sectionMarker =
       (argv.saucelabs || argv.saucelabs_lite) ? 'saucelabs' : 'local';
 
-  let resolver;
-  const deferred = new Promise(resolverIn => {resolver = resolverIn;});
-  new Karma(c, function(exitCode) {
-    if (shouldCollapseSummary) {
-      console./* OK*/log('travis_fold:end:console_errors_' + sectionMarker);
-    }
-    server.emit('kill');
-    if (exitCode) {
-      log(
-          red('ERROR:'),
-          yellow('Karma test failed with exit code ' + exitCode));
+  if (argv.saucelabs || argv.saucelabs_lite) {
+    let batch = 1;
```

Comment:
I don't understand what this block does. Care to add a comment explaining the algorithm?

Reply:
Done!